python/histone_utility.py

```diff
@@ -0,0 +1,342 @@
+
+    It is important to note that you pop() any duplicate IDs from ginfo so that
+    the size can stay consistent and this can be placed back into the dataframe.
+
+    INPUT
+        ginfo_list: list of dictionaries containing gene ids, 
+                    symbols, scores, and query
+    
+    OUTPUT
+        gene: list of just the gene symbols from the list of dictionaries 
+              inputted (any gene that could not be converted 
+              will be given the symbol 'delete')
+              
+    """
+    gene = []
+    for entry in ginfo_list:
+        if entry.get('symbol') != None:
+            gene.append(entry['symbol'])
+        else:
+            gene.append("delete")
+    return gene
+
+
+def tissue_dict(df):
+    """Using a df with number indexing and not gene symbol 
+    indexing (so not pearson_dfs) this will create a dictionary that will 
+    match celllines to their tissue type.
+    
+    This has tissue types split with an underscore so it will remove 
+    the underscore. This is only really required for tissue analysis.
+   
+    INPUT
+        df: dataframe of standard indexing, with columns being celllines and 
+            corresponding tissue (for the data used in the code, the tissue 
+            and cellline is seperated by an underscore)
+    
+    OUTPUT
+        tissue_dict: dictionary of cell line and corresponding tissue
+        cellline_list: list of cell lines in the dataframe
+        
+    """
+    celllines = list(df)
+    celllines.pop(0)
+
+    cellline_list =[]
+    tissue_list = []
+    for cell in celllines:
+        temp = cell.split('_')
+        cell = temp[0]
+        cellline_list.append(cell)
+        tissue = temp[1:]
+        tissue = " ".join(tissue)
+        tissue_list.append(tissue)
+
+    d = {'Celllines': cellline_list, 'Tissue' : tissue_list}
+    cell_df = pd.DataFrame(d)
+    cellline_tissue = cell_df.values
+    tissue_dict ={}
+    for data in cellline_tissue:
+        tissue_dict[data[0]] = data[1]
+    return tissue_dict, cellline_list
+
+
+def normalize(df, delete_column, cell_list):
+    """Takes in a dataframe with columns being cell lines 
+    (with the exception of a column being genes or gene_ids) and normalizes 
+    the expression values.
+    
+    This will delete the desired column and then normalize the values and 
+    fill in the NaN values. It will then format the dataframe to back 
+    to how it originally looked. 
+    
+    INPUT
+    
+        df: dataframe with columns of cell lines (with the exception of a 
+            column being genes or gene_ids)
+        delete_column: column that does not contain expression values
+        cell_list: list of cell lines within dataframe
+    
+    OUTPUT
+        df_normalized: dataframe of normalized expression values and 
+                       NaN values filled   
+                       
+    """
+
+    temp_df = df.copy()
+    
+    delete = temp_df[delete_column]
```

Comment:
can be simplified into 1 line: temp.pop([delete_column])

Reply:
im not sure what you mean by this but I changed the code becuase delete and delete_column are different in this case.

Reply:
`pop` removes the object from the dataframe: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.pop.html